bcip-0004.md

```diff
@@ -0,0 +1,83 @@
+# BCIP-0004 : NFC Payment Solution
+
+```
+Number:  BCIP-0004
+Title:   NFC Payment Solution
+Type:    Informational
+Status:  Draft
+Authors: Blockcore <post@blockcore.net>
+Created: 2023-09-19
+```
+
+## Abstract
+
+This is a proposed solution for NFC payments using Blockcore.
+
+NFC has primarily two modes, read/write data and authentication. The common NFC payment solutions are using the authentication mode, where the NFC card is used to authenticate the user and the payment is done using a different channel. Your regular payment cards from fiat banks are using this mode. The authentication mode is not suitable for Blockcore payments, because it requires a central authority to issue the cards and to verify the authentication.
+
+While such a solution could be implemented, it requires a central authority (custodial wallet service) to issue the cards and to verify the authentication. This is not a solution that Blockcore will support, but it is possible to implement such a solution on top of Blockcore for anyone who wants to do that.
+
+## Motivation
+
+NFC payments are a convenient way to pay for goods and services. It is a technology that is widely adopted and supported by most modern smartphones. It is also a technology that is widely used in payment terminals in stores and restaurants, or for festivals and events.
+
+Making use and adoption of Bitcoin and other cryptocurrencies is a challenge, because it is not widely supported by payment terminals. The NFC payment solution will allow Blockcore to be used for smaller amounts, such as paying for a coffee or a beer at a festival.
+
+## Body
+
+The deployment of NFC payment solution requires the following infrastructure:
+
+- Blockcore Payment Server
+- NFC Payment Terminal
+- NFC Payment Card
+- Blockcore Wallet (optional)
+
+The proposed solution requires the private key to be stored on the physical NFC card. This will be the same private key for all payments, until the user decides to wipe/rewrite their private key. For privacy reasons, it is recommended to wipe the private key after some usage, such as after a day or a week or events.
```

Comment:
This is important because in reality key could leak to merchants every time you pay

Reply:
"Encryption should use strong encryption methods like AES-256 or RSA with a secure key management system. The encryption key should be derived from a user's PIN or passphrase, ensuring that only the card owner can decrypt the private key."

PIN only gives a little more security, but more hassle for the user. They need to enter PIN all the time and in public spaces, the PIN can be observed (user might rely on that PIN for other cards/services).

It primarily stops cloned card attacks, and it might be an option for users when they write to their NFC cards. They can choose to have a PIN or no PIN.

"When a user initiates a transaction with an NFC card, the private key is encrypted with the public key of the selected merchant. This means only that specific merchant can decrypt and use the private key."

This is done before user uses their card, not at the time of usage. So the encryption must happen using their wallet device beforehand.

I will copy the rest of your content into the proposal, thanks!

Reply:
I added some paragraph regarding PIN in the rephrased content, but the rest I can't use as it needs fixes and it's mostly the same as already written.